tests_metricflow/dataflow/optimizer/source_scan/test_source_scan_optimizer.py

```diff
@@ -110,6 +111,9 @@ def visit_add_generated_uuid_column_node(self, node: AddGeneratedUuidColumnNode)
     def visit_join_conversion_events_node(self, node: JoinConversionEventsNode) -> int:  # noqa: D102
         return self._sum_parents(node)
 
+    def visit_join_to_custom_granularity_node(self, node: JoinToCustomGranularityNode) -> int:  # noqa: D102
```

Comment:
@tlento Should the time spine be considered a source node? It isn't for other nodes, so I didn't treat it as such here. But we should update any nodes that pull in the time spine here if it should be

Reply:
As discussed offline.... Maybe?

I think our best path forward is to move off of our current single-pass approach and instead build CTE source nodes in one pass and then layer the metric nodes on top of those. If we do a good job with that this optimizer will no longer be necessary.